images/spire/tests/main.tf

```diff
@@ -19,51 +19,87 @@ data "oci_string" "ref" {
   input    = each.value
 }
 
-resource "random_id" "hex" { byte_length = 4 }
+data "imagetest_inventory" "this" {}
 
-data "oci_exec_test" "helm" {
-  digest      = var.digests["server"] // Not used, but required by the resource.
-  script      = <<EOF
-set -e
+resource "imagetest_harness_k3s" "this" {
+  name      = "spire"
+  inventory = data.imagetest_inventory.this
+}
+
+module "spire_crds" {
+  source = "../../../tflib/imagetest/helm"
+
+  namespace = "spire-system"
+  chart     = "spire-crds"
+  repo      = "https://spiffe.github.io/helm-charts-hardened/"
+}
 
-rand=${random_id.hex.hex}
+module "spire" {
+  source = "../../../tflib/imagetest/helm"
 
-if ! command -v flock; then
-  echo "flock not installed; use \`brew install flock\`"
-  exit 1
-fi
+  name      = "spire"
+  namespace = "spire-system"
+  chart     = "spire"
+  repo      = "https://spiffe.github.io/helm-charts-hardened/"
 
-cat > /tmp/crd-values-$${rand}.yaml <<EOV
-annotations:
-  "helm.sh/resource-policy": "delete"
-EOV
+  values = {
+    spire-server = {
+      enabled = true
+      image = {
+        registry   = data.oci_string.ref["server"].registry
+        repository = data.oci_string.ref["server"].repo
+        tag        = data.oci_string.ref["server"].pseudo_tag
+      }
+    }
+    spire-agent = {
+      enabled = true
+      image = {
+        registry   = data.oci_string.ref["agent"].registry
+        repository = data.oci_string.ref["agent"].repo
+        tag        = data.oci_string.ref["agent"].pseudo_tag
+      }
+    }
+    spiffe-oidc-discovery-provider = {
+      enabled = true
+      image = {
+        registry   = data.oci_string.ref["oidc-discovery-provider"].registry
+        repository = data.oci_string.ref["oidc-discovery-provider"].repo
+        tag        = data.oci_string.ref["oidc-discovery-provider"].pseudo_tag
+      }
+      config = {
+        acme = {
+          tosAccepted = true
+        }
+      }
+    }
+  }
+}
```

Comment:
Might be a good use case for `jsondecode`.

Reply:
as discussed, going to revisit this in a follow up if/when necessary 🙏